### ISSUE TITLE
Debian/Ubuntu package dependency on `python3-virtualenv` is problematic

### ISSUE THREAD

**Issue by gertvdijk** [Yubico/yubikey-manager#412]
- YubiKey Manager (ykman) version: `4.0.2~ppa1~focal1`
- How was it installed?: [Ubuntu PPA Yubico stable](https://launchpad.net/~yubico/+archive/ubuntu/stable)
- Operating system and version: Ubuntu 20.04
- YubiKey model and version: _not relevant, software packaging issue without the need of a Yubikey_
- Bug description summary: The package dependency on `python3-virtualenv` is problematic for Python development on the same machine, please avoid as _hard_ dependency.

Steps to reproduce
---

1. On a clean system **without** `python3-virtualenv` APT package installed, install a recent version of `virtualenv` using `pip` for the system interpreter, e.g. Python 3.8 as shipped with Ubuntu 20.04. At time of writing `pip list` will then show `virtualenv             20.4.4`. Happy user of virtualenv at this point.
2. Install the PPA & `yubikey-manager` package version 4 (`4.0.2~ppa1~focal1` at time of writing).
3. Notice that `python3-virtualenv` from Ubuntu APT repository is installed as a hard dependency of `yubikey-manager`.
4. Try to use the virtualenv package on newer Python versions installed with e.g. pyenv. It's broken, because the virtualenv developer confirms it breakage and unsupported state of having a both versions installed (https://github.com/pypa/virtualenv/issues/1972#issuecomment-705868619). It will fail in subtle ways, see e.g. https://github.com/pypa/pipenv/issues/4663#issuecomment-825978659

TL;DR: As a Python developer I need the latest `virtualenv` on my system, but the installation of `yubikey-manager` from the PPA forces me to install the Ubuntu-provided version. Breakage is due to a bug with `virtualenv` as root cause (https://github.com/pypa/virtualenv/issues/1972#issuecomment-705868619), not directly caused by anything from this repository.

Expected result
---

Either:
- The ability to install `yubikey-manager` from the Yubico stable PPA without a _hard_ dependency on `python3-virtualenv`. E.g. as _recommended_ dependency instead, so that APT allows me to not install it, and use the alternative installed one. Not pretty, but allows for a work-around.
- Get rid of the virtualenv management altogether. Downstream Debian packaging of version 4 manages this without it apparently; see https://packages.debian.org/bullseye/yubikey-manager. A lot prettier, but not sure if feasible from Yubico's perspective.

Actual results and logs
---

Read https://github.com/pypa/pipenv/issues/4663#issuecomment-825978659 for an example how some Python developers will be pulling their hair out for a while after installing `yubikey-manager` on their system.

**Comment by dainnilsson** [comment]
Thanks for bringing this to our attention. Would switching the dependency in the package to `python3-venv` make things better in this regard?
The official Debian package for yubikey-manager does things more correctly, by using other Debian-packaged versions of Python dependencies, but isn't up to date on older versions of Ubuntu. Packaging ykman as a virtualenv allows us to to use the latest Python dependencies even on older distros.
One workaround is to install yubikey-manager using `pip` instead of through apt. On Ubuntu you'll need `swig` and `libpcsclite-dev` from apt for the installation to work, but that will allow you to use any (or no) version of virtualenv.

**Comment by dainnilsson** [comment]
Actually, looking a bit closer, it doesn't look like we need to have a runtime dependency on virtualenv at all. I believe it should only be needed at build time.

**Comment by gertvdijk** [comment]
Thanks for considering this report!

I think you're right, the currently declared runtime dependency on it seems superfluous indeed. Examples on [the dh-virtualenv documentation site](https://dh-virtualenv.readthedocs.io/en/latest/index.html) also do not list this as runtime dependency.

I could have a try building this with a one liner change in debian/control and see and submit a PR and get back to this.

**Comment by dainnilsson** [comment]
I'm actually already testing a build, so please don't waste your time! I appreciate the offer though, and hope to have a new PPA build published in a few days!

**Comment by gertvdijk** [comment]
Thanks again. I was able to build the package with a bit of effort without the declared runtime dependency and it works perfectly fine.

Looking forward to a new release in the PPA some day. 👍🏼

**Comment by dainnilsson** [comment]
New PPA package published (`4.0.2~ppa4`), with no virtualenv dependency!
